crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4166,8 +4176,16 @@ impl<'db> TypeInferenceBuilder<'db> {
                     ) {
                         return symbol_from_bindings(db, bindings).into();
                     }
+
+                    // There are no visible bindings here.
+                    // Don't fall back to non-eager symbol resolution.
+                    if in_eager_context {
+                        in_eager_context &= enclosing_scope_id.is_eager(db);
+                        continue;
```

Comment:
This brings back some logic duplication that we discussed when first adding eager nested scope support (https://github.com/astral-sh/ruff/pull/16079#discussion_r1951742987).  Ideally we could have the return value of `SemanticIndex::eager_bindings` distinguish between "this enclosing scope is no longer part of an eager context" and "it is but there are no bindings for the symbol".  Then we'd have a three-armed `match` statement here (`return`, `continue`, and fall through), and we wouldn't have to track `in_eager_context` both here and in the semantic index builder.